### ISSUE TITLE
az webapp config set WARNING: App settings have been redacted

### ISSUE THREAD

**Issue by rLindorfer** [Azure/azure-cli#27724]
### Describe the bug
Regression: This problem is caused when upgrading az cli from `2.53.0` to `2.53.1`.

### Description
When using `set` commands as described below, a warning message is produced which breaks our build pipelines with `failOnStandardError` set to true in combination with an error exit code.

### Related commands

- az webapp config container set
- az webapp config appsettings set
- az webapp config set

### Errors

 WARNING: App settings have been redacted.

### Issue script & Debug output

WARNING: cli.azure.cli.command_modules.appservice.custom: App settings have been redacted. Use `az webapp/logicapp/functionapp config appsettings list` to view.

### Expected behavior

Configs should be set without any warnings, because it's the natural behavior of setting configs.

### Environment Summary
```
azure-cli                         2.53.1

core                              2.53.1
telemetry                          1.1.0

Extensions:
azure-devops                      0.26.0

Dependencies:
msal                            1.24.0b2
azure-mgmt-resource             23.1.0b2

Python location '/opt/az/bin/python3'
Extensions directory '/opt/az/azcliextensions'

Python (Linux) 3.10.10 (main, Oct 24 2023, 06:10:14) [GCC 11.4.0]
```

**Comment by g4mb10r** [comment]
Do you have integration tests to detect regression?

**Comment by BeigeBadger** [comment]
I ran into this bug today, thanks for raising an issue @rLindorfer! Hopefully it's resolved soon.

**Comment by marrim-myob** [comment]
we ran into the same bug ,since the agents upgraded to new az cli version

**Comment by mheskandari** [comment]
we also have the same issue

**Comment by BeigeBadger** [comment]
An update to my comment above, I got this issue when using the `az functionapp config appsettings set` command.

cc @yonzhan

**Comment by MaxMommersteeg** [comment]
Same issue here when running `az functionapp config appsettings set`.

**Comment by rLindorfer** [comment]
Yes, it's very easy to reproduce this issue, simply call `az functionapp config appsettings set` and set any setting.

Our current workaround is to switch back to `az cli` 2.53 with the following task:
```
- task: Bash@3
  displayName: "Install AZ CLI 2.53.0"
  inputs:
    targetType: 'inline'
    script: |
      pip install azure-cli==2.53.0
```

**Comment by JonSmith-Work** [comment]
Still happening in Azure CLI 2.54.0

**Comment by jrunestone** [comment]
This issue is unfortunately breaking our pipelines as well.

**Comment by 6heads** [comment]
This still exists in 2.55.0.

@yonzhan Any plans on fixing this?

**Comment by segunak** [comment]
> This still exists in 2.55.0.
> 
> @yonzhan Any plans on fixing this?

+1. Any word on a fix for this? We're trying out use of ` --only-show-errors` (details [here](https://learn.microsoft.com/en-us/cli/azure/azure-cli-configuration#cli-configuration-values-and-environment-variables)) on all calls to `az functionapp function keys set` but its rather annoying to have to do so.

**Comment by ErikMogensen** [comment]
The flag `--only-show-errors` also works for the 32-bit version of Azure CLI 2.55.0 for those commands.

**Comment by rLindorfer** [comment]
This means, the only solution is to use the `--only-show-errors` flag?

**Comment by rLindorfer** [comment]
What if we *do not* want to use the `--only-show-errors` flag?

**Comment by vishal-narayanan** [comment]
<img width="889" alt="image" src="https://github.com/Azure/azure-cli/assets/8446877/a1663468-ef79-4328-a5cb-a88ff0e7b89e">
any one knows why it why it changes the value to false while deploying?

**Comment by mheskandari** [comment]
it happens also in az cli 2.62.0

**Comment by jakub-bochenski** [comment]
Is azure-cli no longer maintaned?

This warning is printed even with --output none

So you are warning me that you are now not printing the settings that I requested you not to print?

**Comment by zorbla** [comment]
Still happening with azure-cli 2.64.0.  It's setting values, then prints warning and then it outputs json of the just changed settings with all values being "null" (which is wrong).

**Comment by sguidos** [comment]
Still happening with azure-cli version 2.65.0, in the Azure Web App Console

**Comment by KitKatNarg** [comment]
Hello Folks,

In Azure:

I have been seeing this "App settings have been redacted" when i am trying to run a command in the Azure Cloud shell,
command i ran : 
`az webapp config appsettings set --name kaazingapp --resource-group myResourceGroup --settings DEPLOYMENT_BRANCH='main' `

`kaazingapp is my webapp name`
![Image](https://github.com/user-attachments/assets/f81220f6-45e5-4891-a4e6-3b0b55a5fac4)

Please provide me a solution for this or how can i resolve this issue!

**Comment by danmacode** [comment]
> Hello Folks,
> 
> In Azure:
> 
> I have been seeing this "App settings have been redacted" when i am trying to run a command in the Azure Cloud shell, command i ran : `az webapp config appsettings set --name kaazingapp --resource-group myResourceGroup --settings DEPLOYMENT_BRANCH='main' `
> 
> `kaazingapp is my webapp name`
> 
> ![Image](https://github.com/user-attachments/assets/02a7fb62-d6fe-457a-bce6-fec3b0953aad)
> 
> ![Image](https://github.com/user-attachments/assets/1711f3e9-fde9-4883-a349-e91567f17a97)
> 
> Please provide me a solution for this or how can i resolve this issue!

Try with `--only-show-errors`

```shell
az functionapp config appsettings list --name kaazingapp --resource-group myResourceGroup --only-show-errors -o json
```

**Comment by seligj95** [comment]
I've created a PR to remove the message from the output. You no longer would need the additional parameters to hide the message. This should be deployed with the next CLI version coming out in early April.
